module/os/windows/debug.c

```diff
@@ -125,12 +125,14 @@ printBuffer(const char *fmt, ...)
 	va_list args;
 	va_start(args, fmt);
 	char buf[max_line_length];
-	_snprintf(buf, 18, "%p: ", PsGetCurrentThread());
+	int buf_used;
 
-	int tmp = _vsnprintf_s(&buf[17], sizeof (buf), max_line_length,
-	    fmt, args);
-	if (tmp >= max_line_length) {
-		_snprintf(&buf[17], 17, "buffer too small");
+	buf_used = _snprintf(buf, sizeof(buf), "%p: ", PsGetCurrentThread());
+
+	int tmp = _vsnprintf_s(buf + buf_used, sizeof(buf) - buf_used, _TRUNCATE, fmt, args);
```

Comment:
... but you would need to check for `buf_used` being -1 here, or address of `buf - 1` would not go well.
Now there is no real reason why should ever get -1 tho..  `buf` is plenty big for a single %p. Maybe just throw in a `if (buf_used < 0) return;`

Reply:
Yea, I was thinking about that too, checking for negative values makes sense even if it should not happen there. I'll update it soon.

Reply:
Updated.